src/goInstallTools.ts

```diff
@@ -369,21 +387,37 @@ export function installTools(missing: string[], goVersion: SemVersion): Promise<
 					resolve([...sofar, null]);
 					return;
 				}
-				const args = ['get', '-u', '-v'];
-				if (tool.endsWith('-gomod')) {
-					args.push('-d');
-				}
-				args.push(getToolImportPath(tool, goVersion));
-				cp.execFile(goRuntimePath, args, { env: envForTools }, (err, stdout, stderr) => {
-					if (stderr.indexOf('unexpected directory layout:') > -1) {
-						outputChannel.appendLine(`Installing ${tool} failed with error "unexpected directory layout". Retrying...`);
-						cp.execFile(goRuntimePath, args, { env: envForTools }, callback);
-					} else if (!err && tool.endsWith('-gomod')) {
-						const outputFile = path.join(toolsGopath, 'bin', process.platform === 'win32' ? `${tool}.exe` : tool);
-						cp.execFile(goRuntimePath, ['build', '-o', outputFile, getToolImportPath(tool, goVersion)], { env: envForTools }, callback);
-					} else {
-						callback(err, stdout, stderr);
+				// Install tools in a temporary directory, to avoid altering go.mod files.
+				fs.mkdtemp(path.join(os.tmpdir(), 'go-tools-'), (err, toolsTmpDir) => {
```

Comment:
We use a specific dir inside the tmp dir in this extension. Can you use that here instead?

```suggestion
				fs.mkdtemp(getTempFilePath('go-tools')), (err, toolsTmpDir) => {
```

Reply:
Also, do we need to create this temp dir for each tool? Why not create it once outside the `reduce()` call?